core/src/util/tileID.h

```diff
@@ -16,6 +16,7 @@ struct TileID {
     
     TileID(int _x, int _y, int _z) : x(_x), y(_y), z(_z) {};
     TileID(const TileID& _rhs): x(_rhs.x), y(_rhs.y), z(_rhs.z) {};
+    TileID(const TileID&& _rhs): x(std::move(_rhs.x)), y(std::move(_rhs.y)), z(std::move(_rhs.z)) {};
```

Comment:
Hi, I'm also wondering how this move could work at all :) I would think move can only be used to say that references to heap blocks should be moved from one object to another and invalidated on the former(but I may be missing something)
Whenever a TileID is allocated, be it on the heap or the stack it is a fixed size contiguous region in memory to hold all members, how could parts from one region be said to be *moved* and not copied to the other?